### PR TITLE
added create/update control server

### PR DESCRIPTION
**Pull request by djosborne**

### CODE REVIEW

File: app/cappy/services/cloud_service.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module Cappy
+  module Services
+    # I don't know what this does
+    module CloudService
+      CLOUD_ADDR = 'http://localhost:4567/auth'
```
Comment (djosborne) [inline app/cappy/services/cloud_service.rb]: can this be hardcoded as a constant since the ip of the ccs will be static?

**Comment by djosborne** [comment]
@nahiluhmot

File: app/cappy/services/cloud_service.rb (hunk 1 above)
Comment (nahiluhmot) [inline app/cappy/services/cloud_service.rb]: You should make the `CloudService` a class whose `instance_variables` are the `host` and `port`.

File: Gemfile [hunk 2]
```diff
@@ -9,6 +9,8 @@ gem 'pry', '~> 0.10.1'
 gem 'sinatra-activerecord', '~> 2.0.4'
 gem 'resque', '~> 1.25.2'
 gem 'clockwork', '~> 1.1.0'
+gem 'excon'
+gem 'macaddr'
```
Comment (nahiluhmot) [inline Gemfile]: We'll get the mac address from the command line, so we won't need this gem.

File: app/cappy/services/cloud_service.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+      include Base
+
+      module_function
+
+      def create_control_server(local_port)
```
Comment (nahiluhmot) [inline app/cappy/services/cloud_service.rb]: Instead of having this post the `Mac.addr`, have this method accept the mac address.

**Comment by djosborne** [comment]
@nahiluhmot

File: app/cappy/services/cloud_service.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+module Cappy
+  module Services
+    # I don't know what this does
```
Comment (tlunter) [inline app/cappy/services/cloud_service.rb]: Why this comment?

**Comment by nahiluhmot** [comment]
@djosborne This is coming together nicely. Some comments:

You're creating an `Excon` connection twice with the same `host` and `port`. Why not refactor it into a method like this?

```ruby
def connection
  @connection ||= Excon.new("http://#{host}/#{port}")
end
```

Then you could make requests by using `connection.get(path: 'your_path', ...)`, etc.

Also, the this seems like a strange place to determine the UUID for the `control_server`. Perhaps that should be another argument to `#initialize`.

**Pull request closed without merging**